### PR TITLE
mail: add package clamsmtp

### PR DESCRIPTION
**Pull request by lucize**
Compile/Tested on x86_64 LEDE Reboot (SNAPSHOT, r3640-f229f4a)

Signed-off-by: Lucian Cristian <lucian.cristian@gmail.com>

### CODE REVIEW

File: mail/clamsmtp/files/clamsmtpd.init [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+#
+#    (c) 2017 Lucian Cristian <lucian.cristian@gmail.com>
+#
+###########################################################################
+# CONFIGURATION
+
+# Most configuration options are found in the clamsmtpd.conf file.
+###########################################################################
+# SCRIPT
+
```
Comment (stintel) [inline mail/clamsmtp/files/clamsmtpd.init]: Please drop the commented lines above and fix the style for the copyright line. Example: https://git.lede-project.org/?p=source.git;a=blob;f=package/network/services/uhttpd/files/uhttpd.init;h=6b8be1c7e4a27aa8c91f6c801af0e8c01ba9214c;hb=HEAD

File: mail/clamsmtp/files/clamsmtpd.init [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+START=91
+USE_PROCD=1
+
+PROG=/usr/sbin/clamsmtpd
+
+start_service(){
+        procd_open_instance
+        procd_set_param command $PROG -p /var/clamsmtpd.pid
```
Comment (stintel) [inline mail/clamsmtp/files/clamsmtpd.init]: PID file should go in /var/run

File: mail/clamsmtp/files/clamsmtpd.init [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+        procd_close_instance
+}
+stop_service(){
```
Comment (stintel) [inline mail/clamsmtp/files/clamsmtpd.init]: Please drop this function, it's not needed.

File: mail/clamsmtp/Makefile [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+include $(INCLUDE_DIR)/package.mk
+
+define Package/$(PKG_NAME)
+  SECTION:=mail
+  CATEGORY:=Mail
+  DEPENDS:=+libpthread
+  TITLE:=$(PKG_NAME)
+  URL:=http://thewalter.net/stef/software/clamsmtp/
+  PKGARCH:=all
```
Comment (stintel) [inline mail/clamsmtp/Makefile]: Remove this line. PKGARCH:=all should only be used for platform-independent code.

File: mail/clamsmtp/Makefile [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+
+MAINTAINER:=Lucian Cristian <lucian.cristian@gmail.com>
+PKG_LICENSE:=GPLv2
+
+PKG_FIXUP:=autoreconf
+PKG_BUILD_PARALLEL:=1
+
+include $(INCLUDE_DIR)/package.mk
+
+define Package/$(PKG_NAME)
```
Comment (stintel) [inline mail/clamsmtp/Makefile]: Use clamsmtp instead of the variable for readability.

File: mail/clamsmtp/Makefile [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+  SECTION:=mail
+  CATEGORY:=Mail
+  DEPENDS:=+libpthread
+  TITLE:=$(PKG_NAME)
```
Comment (stintel) [inline mail/clamsmtp/Makefile]: Same here.

File: mail/clamsmtp/Makefile [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+  URL:=http://thewalter.net/stef/software/clamsmtp/
+  PKGARCH:=all
+endef
+
+define Package/$(PKG_NAME)/description
```
Comment (stintel) [inline mail/clamsmtp/Makefile]: Same here.

File: mail/clamsmtp/Makefile [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+	ClamSMTP is an SMTP filter that allows you to check for viruses using the ClamAV anti-virus software.
+endef
+
+define Package/$(PKG_NAME)/install
```
Comment (stintel) [inline mail/clamsmtp/Makefile]: Same here.

File: mail/clamsmtp/Makefile [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+	$(INSTALL_DIR)  $(1)/usr/sbin \
+			$(1)/etc/init.d
+	$(CP) $(PKG_BUILD_DIR)/src/clamsmtpd $(1)/usr/sbin/
+	$(CP) $(PKG_BUILD_DIR)/doc/clamsmtpd.conf $(1)/etc/
+	$(INSTALL_BIN) ./files/clamsmtpd.init $(1)/etc/init.d/clamsmtpd
+endef
+
+$(eval $(call BuildPackage,$(PKG_NAME)))
```
Comment (stintel) [inline mail/clamsmtp/Makefile]: Same here.

**Comment by lucize** [comment]
made the changes
regards

**Pull request merged**